mix.exs

```diff
@@ -152,6 +152,45 @@ defmodule Mix.Tasks.Compile.Tika do
     end
   end
 
+  defp proxy_auth(url) do
+    # url scheme
+    url
+    |> get_scheme
+    |> get_proxy_auth_from_env()
+  end
+
+  defp get_scheme(url) do
+    cond do
+      String.starts_with?(url, "http://") -> :http
+      String.starts_with?(url, "https://") -> :https
+    end
+  end
+
+  defp get_proxy_auth_from_env(:http) do
+    proxy = System.get_env("HTTP_PROXY") || System.get_env("http_proxy")
+    get_proxy_auth_from_proxy_url(proxy)
+  end
+
+  defp get_proxy_auth_from_env(:https) do
+    proxy = System.get_env("HTTPS_PROXY") || System.get_env("https_proxy")
+    get_proxy_auth_from_proxy_url(proxy)
+  end
+
+  defp get_proxy_auth_from_proxy_url(proxy) do
+    # remove trailing / from proxy url
+    proxy = String.trim(proxy, "/")
+    # http://username:password@ip:port
+    regex = Regex.compile!("(http|https)://(?<username>.+):(?<password>.+)@(.+):(.*)")
```

Comment:
Instead of trying to parse the URL with a regular expression, maybe something like (untested):

```elixir
parsed = URI.parse(proxy)
if parsed.userinfo do
  [username, password] = String.split(parsed.userinfo, ":", parts: 2)
  [proxy_auth: {to_charlist(username), to_charlist(password)}]
else
  []
end
```

Reply:
Good suggestion, thanks. I did the changes however forgot to turn off the formatter in my editor. Hence I force pushed to remove that commit and then did another commit with only the relevant changes (without any elixir formatter changes). Should we continue with this pull request or should I send another PR and remove this one?

Reply:
Sorry for the delays - this PR works just fine! Will leave one more review...